vllm/engine/metrics.py

```diff
@@ -336,14 +344,16 @@ def log(self, stats: Stats) -> None:
                 "Avg generation throughput: %.1f tokens/s, "
                 "Running: %d reqs, Swapped: %d reqs, "
                 "Pending: %d reqs, GPU KV cache usage: %.1f%%, "
-                "CPU KV cache usage: %.1f%%",
+                "CPU KV cache usage: %.1f%%, "
+                "Preempted : %d",
```

Comment:
The value of `stats.preempted` is the value of the number preempted on the last iteration.

I don't think we should be printing this to `stdout` since we are already logging a warning when preemption happens.

If you really do want to print this to `stdout` take a look at the `prompt_throughput` and `generation_throughput` for an example of how to keep track of the total numbers of the last `_local_interval_seconds`

Reply:
we can easily print the total number actually. we are already tracking it in scheduler. Let me just print the cumulative total num

Reply:
I don't love that, the info being logged here is "what happened over the past N seconds"

So I would just remove it. We have enough warnings already for preemptions

Reply:
comments are removed!